library/src/interactor_impl.cxx

```diff
@@ -408,6 +401,40 @@ class interactor_impl::internals
     this->VTKInteractor->ExitCallback();
   }
 
+  /**
+   * Run a camera transition animation based on a camera state interpolation function.
+   * The provided function will be called with an interpolation parameter
+   * varying from `0.` for the initial to `1.` for the final state;
+   * it shall return an appropriate linearly interpolated `camera_state_t` for any value in between.
+   */
+  void AnimateCameraTransition(const std::function<camera_state_t(double)>& interpolateCameraState)
```

Comment:
It's usually preferred to use a template parameter instead of `std::function` when it's a private "inlined" function.
See https://rules.sonarsource.com/cpp/tag/bad-practice/RSPEC-5213/

```suggestion
  template<typename Func>
  void AnimateCameraTransition(Func interpolateCameraState)
```